commit 1bf46f6a026cd11da3101c218a9fdd1d1b5b7dd9
Author: Stefan Wahren <stefan.wahren@i2se.com>
Date:   Mon Oct 22 11:09:18 2018 +0200

    staging: bcm2835: Don't probe if no camera is detected
    
    It is a waste of resources to load the camera driver in case there isn't
    a camera actually connected to the Raspberry Pi. This solution also
    avoids a NULL ptr dereference of mmal instance on driver unload.
    
    Fixes: 7b3ad5abf027 ("staging: Import the BCM2835 MMAL-based V4L2 camera driver.")
    Signed-off-by: Stefan Wahren <stefan.wahren@i2se.com>

diff --git a/drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c b/drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c
index 0e6264810ed4..84ca22d7d659 100644
--- a/drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c
+++ b/drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c
@@ -1847,6 +1847,12 @@ static int bcm2835_mmal_probe(struct platform_device *pdev)
 	num_cameras = get_num_cameras(instance,
 				      resolutions,
 				      MAX_BCM2835_CAMERAS);
+
+	if (num_cameras < 1) {
+		ret = -ENODEV;
+		goto cleanup_mmal;
+	}
+
 	if (num_cameras > MAX_BCM2835_CAMERAS)
 		num_cameras = MAX_BCM2835_CAMERAS;
 
@@ -1955,6 +1961,9 @@ static int bcm2835_mmal_probe(struct platform_device *pdev)
 		gdev[i] = NULL;
 	}
 
+cleanup_mmal:
+	vchiq_mmal_finalise(instance);
+
 	return ret;
 }
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1236_1bf46f6a026cd11da3101c218a9fdd1d1b5b7dd9.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1236_1bf46f6a026cd11da3101c218a9fdd1d1b5b7dd9.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
